x-pack/solutions/observability/plugins/inventory/server/routes/entity_definition/get_entity_definitions.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import * as t from 'io-ts';
+import { createInventoryServerRoute } from '../create_inventory_server_route';
+
+export const getEntityDefinitionSourceIndexPatternsByType = createInventoryServerRoute({
+  endpoint: 'GET /internal/inventory/entity/definitions/sources',
+  params: t.type({
+    query: t.type({
+      types: t.string,
+    }),
+  }),
+  options: {
+    tags: ['access:inventory'],
+  },
+  async handler({ context, params, request, plugins }) {
+    const [_coreContext, entityManagerStart] = await Promise.all([
+      context.core,
+      plugins.entityManager.start(),
+    ]);
+    const { types } = params.query;
+    const entityManagerClient = await entityManagerStart.getScopedClient({ request });
+
+    const entityDefinitionIndexPatterns = await Promise.all(
+      types.split(',').map(async (type) => {
```

Comment:
I wonder if there should be an endpoint that accepts a list of types to avoid making N requests to fetch the index pattern for each type individually.

Reply:
I was thinking the same

Reply:
I added a [comment](https://github.com/elastic/kibana/issues/202298#issuecomment-2582270821) in the issue